### DISCUSSION TITLE
MirrorMaker2 | Replicate  data from topic name "A" from source azure event hub  to topic name "B" in destination Kafka cluster

### DISCUSSION THREAD

**Discussion by Sharad-SKJ052** [strimzi/strimzi-kafka-operator#8522] [General]
Hi Team ,

We are using strimzi mirrormaker2 to replicate data from source azure event hub to destination Kafka cluster. We have a  use case where we need to replicate data  from topic say  "topic_name " to a different topic name with some prefix and postfix "X.topic_name.Y"  in destination Kafka cluster . Is there any policy available where we can replicate data from azure event hub to Kafka cluster and rename the topic in strimzi  mirrormaker2.

Can you help me to figure out how we can solve this use case?

**Comment by scholzj** [comment]
You can try to write your own replication policy. But I do not think it was designed for any arbitrary topic renaming like this, so not sure if it will really work or how to do it.

**Reply by Sharad-SKJ052** [reply]
HI @scholzj ,

How can we integrate our own custom replication policy with strimzi mm2.
If renaming is not possible, is it possible to replicate to some other topic name in destination cluster with custom policy.
Can you help us with with some guide to use our own custom policy in strimzi mm2

**Reply by scholzj** [reply]
I cannot help you with any guide because I have not done anything like that. In general, you would need to create your own Java project which implements the replication policy interface nd does the renaming as you need it. Then you build a JAR from it and use a Dockerfile to build a custom container image where you start from the Strimzi image and add your JAR to the classpath. And then you use the image in your custom resource.
